### ISSUE TITLE
FR:  Add support for Debian 12

### ISSUE THREAD

**Issue by pyrodie18** [ansible-collections/community.zabbix#1005]
##### SUMMARY
Debian 12 gets released on June 10th.  I fully expect Zabbix will start to support it shortly thereafter.
 
##### ISSUE TYPE
- Feature Idea

**Comment by Thulium-Drake** [comment]
It might be useful to also consider changing the format from ```.list``` to ```.sources```. In my other playbooks I use the following task:

```
- name: 'Configure Debian main repos'
  # yamllint disable rule:line-length
  ansible.builtin.copy:
    content: |
      # This file is managed by Ansible.
      # YOUR CHANGES WILL BE LOST!
      Types: deb
      URIs: {{ debian_repo_mirror }}/debian
      Suites: {{ ansible_facts['distribution_release'] }} {{ (debian_testing) | ternary('', ansible_facts['distribution_release'] + '-updates') }} {{ (debian_backports_enabled) | ternary(ansible_facts['distribution_release'] + '-backports', '') }}
      Components: {{ debian_repo_components }}

      Types: deb
      URIs: {{ debian_repo_mirror }}/debian-security
      Suites: {{ ansible_facts['distribution_release'] }}-security
      Components: {{ debian_repo_components }}
    dest: '/etc/apt/sources.list.d/debian.sources'
    owner: 'root'
    group: 'root'
    mode: '0644'
  notify: 'Update apt cache'
```

**Comment by pyrodie18** [comment]
> It might be useful to also consider changing the format from `.list` to `.sources`. In my other playbooks I use the following task:
> 
> ```
> - name: 'Configure Debian main repos'
>   # yamllint disable rule:line-length
>   ansible.builtin.copy:
>     content: |
>       # This file is managed by Ansible.
>       # YOUR CHANGES WILL BE LOST!
>       Types: deb
>       URIs: {{ debian_repo_mirror }}/debian
>       Suites: {{ ansible_facts['distribution_release'] }} {{ (debian_testing) | ternary('', ansible_facts['distribution_release'] + '-updates') }} {{ (debian_backports_enabled) | ternary(ansible_facts['distribution_release'] + '-backports', '') }}
>       Components: {{ debian_repo_components }}
> 
>       Types: deb
>       URIs: {{ debian_repo_mirror }}/debian-security
>       Suites: {{ ansible_facts['distribution_release'] }}-security
>       Components: {{ debian_repo_components }}
>     dest: '/etc/apt/sources.list.d/debian.sources'
>     owner: 'root'
>     group: 'root'
>     mode: '0644'
>   notify: 'Update apt cache'
> ```

Thanks for the idea @Thulium-Drake .  I will admit, I'm much more familiar with RHEL vs Debian repos.  What is the difference between .list and .sources?

**Comment by Thulium-Drake** [comment]
Well, the full description is here: https://repolib.readthedocs.io/en/latest/deb822-format.html

TL;DR, One-line is supposed to be deprecated in the (near?) future, Deb822 format is the way to go. I think it's related to the deprecation of ```apt-key```. As you're now supposed to save your GPG keys and/or keyrings in ```/etc/apt/keyrings``` and use something like this:

```
deb [signed-by=/usr/share/keyrings/deriv-archive-keyring.pgp] https://deriv.example.net/debian/ stable main
```

Or in Deb822 format:

```
Types: deb
URIs: https://deriv.example.net/debian/
Suites: stable
Components: main
Signed-By: /usr/share/keyrings/deriv-archive-keyring.pgp
```

And I guess I understand, especially with all the other possible options, the one-line format is getting a bit cramped, and the multiline format allows for clearer management of sources. Though I'm not sure _when_ APT started supporting this.

**Comment by pyrodie18** [comment]
Got it.  OK.  I agree that this is probably a good change to do, but it falls outside of the scope of this ticket.  I'd suggest you go ahead and essentially copy and paste your initial and follow-up posts into a new ticket that we can track appropriately.

**Comment by Thulium-Drake** [comment]
Done! Thanks! ;-)
